### PR TITLE
Issue#1: add backend Add-Function and React-Router

### PR DESCRIPTION
**Pull request by Anmavel**

### CODE REVIEW

File: backend/src/main/java/com/example/backend/model/PetDTO.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.example.backend.model;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public record PetDTO(String name,
+                     String nameOfBreed,
+                     String photo,
+                     String[] supplies) {
```
Comment (MoinMarcell) [inline backend/src/main/java/com/example/backend/model/PetDTO.java:9]: You can use a List and than you dont neet to override methods ;-)
```suggestion
                     List<String> supplies) {
```

File: backend/src/main/java/com/example/backend/model/Pet.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.Objects;
+
+@Document("pets")
+public record Pet(
+        @Id
+        String id,
+        String name,
+        String nameOfBreed,
+        String photo,
+        String[] supplies) {
```
Comment (MoinMarcell) [inline backend/src/main/java/com/example/backend/model/Pet.java:15]: You can use a List here, than you dont need to override methods ;-)

```suggestion
        List<String> supplies) {
```

File: backend/src/main/java/com/example/backend/model/Pet.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        @Override
+        public String toString() {
+                return "Pet{" +
+                        "id='" + id + '\'' +
+                        ", name='" + name + '\'' +
+                        ", nameOfBreed='" + nameOfBreed + '\'' +
+                        ", photo='" + photo + '\'' +
+                        ", supplies=" + Arrays.toString(supplies) +
+                        '}';
+        }
```
Comment (MoinMarcell) [inline backend/src/main/java/com/example/backend/model/Pet.java:45]: Not neccessary if using a List

File: backend/src/test/java/com/example/backend/model/PetDTOTest.java
Comment (MoinMarcell) [inline backend/src/test/java/com/example/backend/model/PetDTOTest.java:1]: No need to test the DTO

File: backend/src/test/java/com/example/backend/model/PetTest.java
Comment (MoinMarcell) [inline backend/src/test/java/com/example/backend/model/PetTest.java:1]: No need to test the Model

File: backend/src/test/java/com/example/backend/service/PetServiceTest.java
Comment (MoinMarcell) [inline backend/src/test/java/com/example/backend/service/PetServiceTest.java:1]: You should rename your tests like when...._then....

**Review by MoinMarcell — CHANGES_REQUESTED** [review]
You only code the backend. Where is the frontend? You are testing your models but why? 
Only 50% of the vertical slice. You need to do major changes.

File: backend/src/main/java/com/example/backend/model/PetDTO.java (hunk 1 above)
Comment (Anmavel) [reply backend/src/main/java/com/example/backend/model/PetDTO.java:9]: Hallo Marcell. Danke fürs Review. Ich wollte nicht zu viele Zeilen review lassen und obwohl ich 100% test Coverage bei Service und Controller hatte, hat SonarCloud gefordert Pet und PetDTO zu testen um diese Mindest-Test-Coverage zu erreichen. Danke dir für das Review :-)

File: backend/src/test/java/com/example/backend/controller/PetControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    @BeforeEach
+    void setUp(){
+        List<String> supplies = new ArrayList<>(List.of("Water Bottle","Roomy Cage"));
+        pet1 =new Pet("1","Whiskers","albino","albino.png",supplies);
+    }
+
+    @Test
+    @DirtiesContext
+    void addTask() throws Exception {
```
Comment (MoinMarcell) [inline backend/src/test/java/com/example/backend/controller/PetControllerTest.java:40]: Rename your test to when..._then...

File: backend/src/test/java/com/example/backend/service/PetServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        idService=mock(IdService.class);
+        petService=new PetService(petRepo,idService);
+        List<String> supplies = new ArrayList<>(List.of("Water Bottle","Roomy Cage"));
+        pet1DTO =new PetDTO("Whiskers","albino","albino.png",supplies);
+        pet1 =new Pet("1",pet1DTO.name(),pet1DTO.nameOfBreed(), pet1DTO.photo(), pet1DTO.supplies());
+
+    }
+
+    @Test
+    void addPet() {
```
Comment (MoinMarcell) [inline backend/src/test/java/com/example/backend/service/PetServiceTest.java:38]: same here

File: frontend/src/component/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+type HeaderProps={
+    image:string|undefined
+}
+export default function Header(props:HeaderProps){
+
+    return(
+
+        <header className={"header"}>
+            <img src={props.image} alt={"logo Petfit"}/><br/>
```
Comment (MoinMarcell) [inline frontend/src/component/Header.tsx:11]: the image could be undefined. you better check this before

File: frontend/src/page/AddPet.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    return (
+        <form onSubmit={formSubmitHandler} className={"add-pet"}>
+            <label>
+                <h2>Please write the name of your Pet</h2>
+                <input type={"text"} onChange={handleNameChange} value={name} placeholder={"write the name of your Pet"}
+                       required={true}/>
+                <input type={"text"} value={nameOfBreed} placeholder={"breed"} required={false}/>
+                <input type={"text"} value={photo} placeholder={"photo"} required={false}/>
+                <input type={"text"} value={supplies} placeholder={"water bottle"} required={false}/>
```
Comment (MoinMarcell) [inline frontend/src/page/AddPet.tsx:48]: where are the onChange methods? I can now only change the name

**Review by MoinMarcell — CHANGES_REQUESTED** [review]
I can only edit the name of the pet in the AddPet Component. Did you test it by yourself?

**Review by Anmavel — COMMENTED** [review]
Thank you Marcell

**Review by Anmavel — COMMENTED** [review]
Thanl you Marcell. I already applied the changes

**Review by MoinMarcell — APPROVED** [review]
lgtm :-)

**Pull request merged**